apps/chat/src/components/Chat/ChatMessage/ChatMessageContent/UserMessage.tsx

```diff
@@ -0,0 +1,553 @@
+      handleToggleEditing(false);
+    },
+    [
+      message,
+      messageContent,
+      handleToggleEditing,
+      conversation,
+      onEdit,
+      newEditableAttachments,
+      selectedDialLinks,
+      messageIndex,
+    ],
+  );
+
+  const handlePressEnter = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === 'Enter' && !isTyping && !e.shiftKey) {
+      e.preventDefault();
+      handleEditMessage();
+    }
+  };
+
+  const handleUnselectFile = useCallback(
+    (fileId: string) => {
+      dispatch(FilesActions.uploadFileCancel({ id: fileId }));
+      const fid = isFolderId(fileId) ? fileId.slice(0, -1) : fileId;
+      setNewEditableAttachmentsIds((ids) => ids.filter((id) => id !== fid));
+    },
+    [dispatch],
+  );
+
+  const handleRetry = useCallback(
+    (fileId: string) => {
+      return () => dispatch(FilesActions.reuploadFile({ fileId }));
+    },
+    [dispatch],
+  );
+
+  const handleSelectAlreadyUploaded = useCallback((result: unknown) => {
+    if (typeof result === 'object') {
+      const selectedFilesIds = result as string[];
+      const uniqueFilesIds = uniq(selectedFilesIds);
+      setNewEditableAttachmentsIds(
+        uniqueFilesIds.map((id) => (isFolderId(id) ? id.slice(0, -1) : id)),
+      );
+    }
+  }, []);
+
+  const handleUploadFromDevice = useCallback(
+    (
+      selectedFiles: Required<Pick<DialFile, 'fileContent' | 'id' | 'name'>>[],
+      folderPath: string | undefined,
+    ) => {
+      selectedFiles.forEach((file) => {
+        dispatch(
+          FilesActions.uploadFile({
+            fileContent: file.fileContent,
+            id: file.id,
+            relativePath: folderPath,
+            name: file.name,
+          }),
+        );
+      });
+
+      setNewEditableAttachmentsIds((ids) =>
+        uniq(ids.concat(selectedFiles.map(({ id }) => id))),
+      );
+    },
+    [dispatch],
+  );
+
+  const handleToggleEditingTemplates = useCallback(
+    (value?: boolean) => {
+      toggleEditingTemplates(value ?? !isEditingTemplates);
+    },
+    [isEditingTemplates, toggleEditingTemplates],
+  );
+
+  useEffect(() => {
+    setMessageContent(message.content);
+  }, [message.content]);
+
+  useEffect(() => {
+    if (message.custom_content?.form_value) {
+      setFormValue(message.custom_content.form_value);
+    }
+  }, [message]);
+
+  useEffect(() => {
+    const links = getDialLinksFromAttachments(
+      message.custom_content?.attachments,
+    );
+    setSelectedDialLinks(links);
+  }, [message.custom_content?.attachments]);
+
+  useEffect(() => {
+    setNewEditableAttachmentsIds(mappedUserEditableAttachmentsIds);
+  }, [mappedUserEditableAttachmentsIds]);
+
+  useEffect(() => {
+    if (isEditing) {
+      setShouldScroll(true);
+    }
+  }, [isEditing]);
+
+  useEffect(() => {
+    if (shouldScroll) {
+      anchorRef.current?.scrollIntoView({ block: 'end' });
+      setShouldScroll(false);
+    }
+  }, [shouldScroll]);
+
+  return isEditing ? (
```

Comment:
use `if` instead. It will be more readable (I couldn't understand why two `UserSchema` are used here)
```suggestion
  if(isEditing) {
       return <... />
  }
  return <... />
```